src/module.c

```diff
@@ -837,11 +837,32 @@ static void GetRedisVersion() {
   if (!reply || RedisModule_CallReplyType(reply) == REDISMODULE_REPLY_ERROR) {
     isCrdt = false;
   }
-
   if (reply) {
     RedisModule_FreeCallReply(reply);
   }
 
+  isMaster = true;
```

Comment:
Role can change during the processes lifetime so you can not count on this variable to be up to date